src/autoscaler/scalingengine/scalingengine.go

```diff
@@ -72,7 +72,7 @@ func (s *scalingEngine) Scale(appId string, trigger *models.Trigger) (int, error
 	if err != nil {
 		logger.Error("failed-to-get-app-info", err)
 		history.Status = models.ScalingStatusFailed
-		history.Error = "failed to get app info"
+		history.Error = "failed to get app info: " + err.Error()
```

Comment:
given the history will be exposed to user, not sure whether it is good to show the detailed / internal errors

Reply:
Yes, similar concern.. Autoscaler cli display every field of scaling history as a column in table. Then, with the detailed error msg, the error column will be quite long ..

Reply:
@boyang9527  Scaling can fail for many reasons, user should be informed about the reason so that they can take necessary action to resolve such failure if possible. ex. failure due to any quota related issue like SpaceQuotaMemoryLimitExceeded or SpaceQuotaInstanceMemoryLimitExceeded or SpaceQuotaTotalRoutesExceeded. Until and unless user is not informed about the root cause of such failure it will be difficult to handle such.

Reply:
@boyang9527  It makes sense not to expose all the errors ( specially the internal errors) to the users. So updated the PR by removing the internal errors.